### PR TITLE
Introduced two version support feature

### PR DESCRIPTION
**Pull request by hlts2**
Now that Vald can support multiple versions, clients should also be able to support multiple versions.
In this PR, I implemented a workflow to support multiple versions.
The main logic is implemented as a common workflow in [vald-client-ci](https://github.com/vdaas/vald-client-ci), so I implemented a call to it.

## Summary by CodeRabbit

- **New Features**
  - Introduced workflows for end-to-end testing and syncing Vald using GitHub Actions.
  
- **Chores**
  - Updated licensing information in the release workflow.
  - Added new targets in the Makefile for managing dependencies, testing, and publishing.
  
- **Documentation**
  - Updated the `NODE_VERSION` to `22` in the new NODE_VERSION file.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,133 +1,32 @@
...
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          tag_name: ${{ github.ref }}
-          release_name: Release ${{ github.ref }}
-          draft: false
-          prerelease: false
+    # TODO: change to the following before main merge.
+    # uses: vdaas/vald-client-ci/.github/workflows/_release.yaml@main
+    uses: vdaas/vald-client-ci/.github/workflows/_release.yaml@feat/add-common-workflows
```
Comment (hlts2) [inline .github/workflows/release.yml:29]:
```suggestion
    # uses: vdaas/vald-client-ci/.github/workflows/release.yaml@main
    uses: vdaas/vald-client-ci/.github/workflows/release.yaml@feat/add-common-workflows
```

**Comment by hlts2** [comment]
@coderabbitai review

File: Makefile [hunk 2]
```diff
@@ -216,30 +230,69 @@ vald/sha/print:
...
-vald/sha/update: vald
-	(cd vald; git rev-parse HEAD > ../$(VALD_SHA))
+vald/sha/update: $(VALD_DIR)
+	(cd $(VALD_DIR); git rev-parse HEAD | tr -d '\n' > ../$(VALD_SHA))
 
-.PHONY: vald/client/node/version/print
-## print VALD_CLIENT_NODE_VERSION value
-vald/client/node/version/print:
+.PHONY: vald/client/version/print
+## print VALD_CLIENT_JAVA_VERSION value
```
Comment (datelier) [inline Makefile:237]:
```suggestion
## print VALD_CLIENT_NODE_VERSION value
```

**Review by datelier — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (hlts2) [reply Makefile:237]: Thank you for catching it! 🙏  I apply your suggestion!

**Review by datelier — APPROVED** [review]

**Pull request merged**